gk/main.c

```diff
@@ -542,6 +604,42 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	return ret;
 }
 
+static int
+drop_flow_entry_randomly(
+	struct gk_instance *instance, unsigned int flow_ht_size)
+{
+	int  ret;
+	uint32_t entry_id = rte_rand() % flow_ht_size;
+	uint32_t entry_start_idx = entry_id;
+	struct flow_entry *fe;
+
+	while (true) {
+		fe = &instance->ip_flow_entry_table[entry_id];
+		if (fe->last_active_ts == 0) {
+			entry_id = (entry_id + 1) % flow_ht_size;
+			fe = NULL;
+		} else
```

Comment:
Combined with the comment below, I think we can make this simpler. Make the if statement test for the opposite case, so that if the last_active_ts != 0 there is a break. Otherwise, just update entry_id and test if it’s equal to entry_start_idx.